### DISCUSSION TITLE
Improvements suggestions

### DISCUSSION THREAD

**Discussion by tedo-develop** [shlinkio/discussions#36] [Ideas]
Hello acelaya,

we like shlink and using it for our company.

We have more than 2000 short urls and over 2500 tags on our installation (works perfect).

Our employees asked us for some improvement and gave us a list with suggestions. 
We hope the suggestions are interesting for you.

Creating Shorturl
- Add multiple tags via comma separated list e.g. newsletter,edition-1.....

Managed-Tags
- Loading page via ajax (infinity scroll) or implement of pagination
- Search tags by full text search
- Show full name of long tags by hover (or something similar)

Statistics of redirects
- Add functionality of clearing stats of an short url

Qr code
- Is it possible to set default settings for qr codes in .env file? Like margin or default file type (png/svg).

Thanks for your work!

**Comment by acelaya** [comment]
Thanks for the suggestions!

> Add multiple tags via comma separated list e.g. newsletter,edition-1

Can you elaborate on this? Do you mean that users want to paste the whole list of comma-separated tags in one go?

> Loading page via ajax (infinity scroll) or implement of pagination.

Tracked: https://github.com/shlinkio/shlink-web-client/issues/459

> Search tags by full text search

This is already supported, unless I'm missing something.

The tags section has a search bar where you can write arbitrary text, and it filters the list of tags with those matching the search term.

> Show full name of long tags by hover (or something similar)

Tracked: https://github.com/shlinkio/shlink-web-client/issues/460

> Add functionality of clearing stats of an short url

Tracked: https://github.com/shlinkio/shlink/issues/1148

It also relates to this other request: https://github.com/shlinkio/shlink/issues/1110

> Is it possible to set default settings for qr codes in .env file? Like margin or default file type (png/svg).

This is something I have considered myself.

Tracked: https://github.com/shlinkio/shlink/issues/1149

**Reply by tedo-develop** [reply]
Hello Acelaya,

i try to explain our suggestions :).

> Can you elaborate on this? Do you mean that users want to paste the whole list of comma-separated tags in one go?

Yes like **newsletter,company1,advertising**

> This is already supported, unless I'm missing something.

Yes if you search on the **manage tags** page but the search on tab **overview** works only with "starts with". I have attached two screenshots from one of our instances.

![Screenshot 2021-08-12 at 07 44 48](https://user-images.githubusercontent.com/39614855/129145124-9ea7cd8b-0611-4ac6-b20d-1fb7edb6f628.png)
![Screenshot 2021-08-12 at 07 45 08](https://user-images.githubusercontent.com/39614855/129145129-1dc450af-61ba-443e-acf5-3c3a00227b7a.png)
![Screenshot 2021-08-12 at 07 49 07](https://user-images.githubusercontent.com/39614855/129145131-b4bba995-2de4-4fd4-9351-ec3630559c17.png)

**Reply by acelaya** [reply]
Ah, ok. Got it now.

Both tracked:

* https://github.com/shlinkio/shlink-web-client/issues/462
* https://github.com/shlinkio/shlink-web-client/issues/463

**Reply by acelaya** [reply]
I have already implemented some of the suggestions related with tags. Will come to shlink-web-client 3.3.0

* Titles for too long tags: https://youtu.be/YU8LgAOqBlE
* Create multiple comma-separated tags at once: https://youtu.be/YcztrLRGYR8
* Suggest tags that "include" the input text, instead of those that "start with" the input text: https://youtu.be/2JaPCqkeBEE

The three of them can be tested in this sandbox, in case you want to test them and give some feedback: https://shlinkio.github.io/shlink-web-client/feature-tags-global-search/

**Reply by tedo-develop** [reply]
Hello Acelaya,

sorry for the late response. But i have good news, we love it!

We have tested the new version and found a new feature.
If you search on page "manage tags" for example sigmatek-gmbh-&-co-kg you will get the correct tag. 
But if you click on "Short-URLs" the tag los any chars after "&". The function "Visits" works as expected.

1. Search works
![Screenshot 2021-08-31 at 08 46 31](https://user-images.githubusercontent.com/39614855/131457397-33ee009c-6c23-43b8-bab7-1eefb04d548b.png)

2. Click on Short-URLs
![Screenshot 2021-08-31 at 08 46 38](https://user-images.githubusercontent.com/39614855/131457437-96410f18-537a-485f-87d6-ddfff61fcef3.png)

3. No Results
![Screenshot 2021-08-31 at 08 46 45](https://user-images.githubusercontent.com/39614855/131457464-a3c3588d-33a2-43b3-ae9d-3bbe4dd222ea.png)

Thanks for your work!

**Reply by acelaya** [reply]
Can you open a separated ticket for that?

Thanks!
